mobile-widgets/qmlmanager.cpp

```diff
@@ -108,6 +109,24 @@ extern "C" int gitProgressCB(const char *text)
 	return 0;
 }
 
+// this callback is used from the uiNotification() function
+// the detour via callback allows us to keep the core code independent from QMLManager
+// I'm not sure it makes sense to have three different progress callbacks,
+// but the usage models (and the situations in the program flow where they are used)
+// are really vastly different...
+// this is mainly intended for the early stages of the app so the user sees that
+// things are progressing
+extern void showProgress(QString msg)
+{
+	static QMLManager *self;
```

Comment:
There appears to be no reason to make this a static variable. This function should probably read as
```
        QMLManager *self = QMLManager::instance();
        if (self)
                self->setNotificationText(msg);
```

Reply:
we have that same code a few lines above. The idea was to not constantly call the `instance()` method and instead just cache the value locally. But yes, you are of course correct. No reason to do this.

Editorial comment... I always find it interesting to see where people worry about wasting cycles and where they don't care. And what patterns are causing them to look for alternatives. Nothing wrong with any of the approaches, it's just interesting how you and I get 'triggered' by very different patterns :-)

Reply:
> The idea was to not constantly call the instance() method and instead just cache the value locally. But yes, you are of course correct. No reason to do this.

Note that this strategy will not help: Since compilers do whole-translation unit optimizations, they will simply inline that `instance()` method, which is a trivial access to a global variable.

OTOH, the version with the local static variable might do some locking, so it might actually be less effective. In this case it probably won't do locking because you are const-initializing the variable and then these things behave like a "global" static. Yes, C++ is complicated. :-P

Reply:
fixed both instances of this and decided to have one call the other because... why not.
which immediately showed another bug that I'm not tracking down...
we get a stray notification that we should get during initialization, and clearly libgit isn't doing a good job sending us update notification as we open local and remote repos.